Remotion/Data/DomainObjects.UnitTests/Queries/QueryFactoryTest.cs

```diff
@@ -330,7 +330,44 @@ public void CreateLinqQuery_WithoutParserAndExecutor ()
       }
     }
 
-    private void ResetCaches ()
+    [Test]
+    public void CreateLinqQuery_WithoutMetadata_UsesEmptyMetadata ()
+    {
+      var factoryMock = new Mock<ILinqProviderComponentFactory>(MockBehavior.Strict);
+      var serviceLocator = DefaultServiceLocator.Create();
+      serviceLocator.RegisterSingle(() => factoryMock.Object);
+      RegisterStandardConfiguration(serviceLocator);
+
+      using (new ServiceLocatorScope(serviceLocator))
+      {
+        var fakeExecutor = new Mock<IQueryExecutor>();
+        var fakeQueryParser = new Mock<IQueryParser>();
+        var fakeResult = new Mock<IQueryable<Order>>();
+
+        factoryMock
+            .Setup(mock => mock.CreateQueryExecutor(
+                TestDomainStorageProviderDefinition,
+                "dummyID",
+                It.Is<IReadOnlyDictionary<string, object>>(d => d.Count == 0)))
+            .Returns(fakeExecutor.Object)
+            .Verifiable();
+        factoryMock
+            .Setup(mock => mock.CreateQueryParser())
+            .Returns(fakeQueryParser.Object)
+            .Verifiable();
+        factoryMock
+            .Setup(mock => mock.CreateQueryable<Order>(fakeQueryParser.Object, fakeExecutor.Object))
+            .Returns(fakeResult.Object)
+            .Verifiable();
+
+        var result = QueryFactory.CreateLinqQuery<Order>("dummyID");
+
+        factoryMock.Verify();
```

Comment:
Can't we convert `factoryMock` to a stub? We get back a result that only works if all setups were called because each setup returns an object that is necessary for the next call, resulting in the final result object that we assert below.

Reply:
since we specifically want to check that the mock is called with empty metadata it can't really be a stub

Reply:
The return values do check this implicitly, so we don't need the mock-verification. 

The second question is: the same test we already have so what's the benefit of "cleaning up" the semantics in this test. 

I can live with both versions.